README.md

```diff
@@ -1,133 +1,112 @@
-- The RedisGears functions
+## Architecture
+### RedisAI
+Redis instance is launched with RedisAI module loaded (by running `redislabs/redisai:latest` docker). Recall that RedisAI allows you to store and execute AI/ML pre-trained models in Redis.
+
+### Data Loader
+Historical data of credit card transactions is loaded into Redis from `data/creditcard.csv` file, using redis-py (Redis' official python client). Every transaction is stored in Redis as a hash with the following structure:
+- The key is `<time_stamp>_<index>{tag}`, where `<index>` is used to differentiate between keys which correspond to transactions that have occurred in the same timestamp, and `{tag}` is used to ensure that all hashes are stored in the same shard (when using clustered environment). 
+- The hash value is a dictionary that contains 29 numeric features ("v0", "v1", ... "v28") that represent the transaction, and another field ("amount") specifying the amount of money to transfer in the transaction.
+
+In addition, all the hashes' keys are kept in a sorted set stored in Redis whose key is `refernces{tag}`, and every element in it is of the form: `{hash_key_name: timestamp}`. This sorted set keeps track of the keynames of these hashes sorted by time.
 
-The `dataloader` container will [load](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py)
-- 1000 [raw reference data points](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py#L29)
-- 1000 [tensor](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py#L32) representations of these raw data points
-- 1 [sorted set](https://github.com/RedisAI/FraudDetectionDemo/blob/master/dataloader/load.py#L35)
+### App
+The app is a basic component that loads the fraud detection pre-trained ML model into RedisAI, along with a [TorchScript](https://oss.redis.com/redisai/intro/#scripting) which is used for pre-processing of the data. This model was built using Tensorflow, which is one of the three ML frameworks that RedisAI supports as backends.
 
-### Explore loaded reference data
-This demo bundles a [RedisInsight](https://redislabs.com/redisinsight/) container which is an intuitive GUI for Redis. Open a browser and point it at https://localhost:8001 and select the preloaded redis connection.
+**Multiple devices:** RedisAI allows executing operation in parallel on different logical devices. Hence, the fraud-detection model is loaded twice under two different key names: `fraud_detection_model{tag}_CPU` and `fraud_detection_model{tag}_CPU:1`, which are associated to `CPU` and `CPU:1` devices respectively. Side note - If your machine has GPU with nvidia CUDA support, it is a good practice to load models that can run in parallel and associate each one with a different device (for example: `fraud_detection_model{tag}_GPU:0` and `fraud_detection_model{tag}_GPU:1`), for gaining better utilization of GPU resources.
```

Comment:
can you re write to something like:
1. `fraud_detection_model{tag}_CPU` will be saved on `CPU`
2. `fraud_detection_model{tag}_CPU:1` will be saved on `CPU:1`

also, for using GPU:0, GPU:1 you will need a multiple GPU machine so the last sentence is not accurate

Reply:
OK, I will rephrase

Reply:
minor rephrase **Multiple devices** RedisAI allows parallel execution of models on different logical devices....In this demo, the fraud detection model is loaded on 2 logical devices